lucene/core/src/java/org/apache/lucene/index/OrdinalMap.java

```diff
@@ -271,13 +273,26 @@ protected boolean lessThan(TermsEnumIndex a, TermsEnumIndex b) {
       globalOrd++;
     }
 
-    this.firstSegments = firstSegments.build();
-    this.globalOrdDeltas = globalOrdDeltas.build();
+    long ramBytesUsed = BASE_RAM_BYTES_USED + segmentMap.ramBytesUsed();
+    this.valueCount = globalOrd;
+
+    // If the first segment contains all of the global ords, then we can apply a small optimization
+    // and hardcode the first segments and global ord deltas as all zeroes.
```

Comment:
Insert possessive quote (`first segment's`)?

Reply:
I don't think the possessive quote gives the right meaning? Perhaps I could say 'first segment indices' here to be more clear.

Reply:
Oh I thought it was `the first segment's deltas as all zeros` and alos the `global ord deltas as all zeros`?  But I'm OK with just rewording it to make it less controversial, or even just leaving this wording!